packages/components/pie-modal/src/index.ts

```diff
@@ -134,7 +149,13 @@ export class PieModal extends RtlMixin(LitElement) implements ModalProps {
 
         if (previousValue !== undefined) {
             if (previousValue) {
-                this._dispatchModalCustomEvent(ON_MODAL_CLOSE_EVENT);
+                if (this._backButtonClicked) {
+                    // Reset the flag
+                    this._backButtonClicked = false;
+                    this._dispatchModalCustomEvent(ON_MODAL_BACK_EVENT);
+                } else {
+                    this._dispatchModalCustomEvent(ON_MODAL_CLOSE_EVENT);
+                }
```

Comment:
This totally isn't caused from your changes on their own, but I think they just exacerbated my confusion reading it!

I think `previousValue` isn't very clear. It should likely be called something more descriptive. I'm guessing it gets the current (changed) value of `isOpen`, but I'm not entirely sure tbh.

Either a name change, or some comments would definitely help (if you know a bit more about the context here.

Reply:
They are the values the properties were _before_ the update cycle. I think this is actually less confusing. How about `oldValue`?

Reply:
I've added a comment, let me know if its any more useful :)